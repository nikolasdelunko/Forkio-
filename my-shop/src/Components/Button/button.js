import './Button.scss'
import React from 'react';

const Button = (props) => {
    const {classN, text, disabled, onClick} = props
    return (
        <div className='btn-container'>
            <button
                className={classN}
                onClick={onClick}
                disabled={disabled}
            >{text}</button>
        </div>
    );
};

export default Button;
