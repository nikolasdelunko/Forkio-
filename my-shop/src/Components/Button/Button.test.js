import { shallow } from "enzyme";
import * as React from "react";
import Button from "./Button";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Button Component", () => {
    it("Should render correctly in DOM", () => {
        const btn = shallow(<Button />);
        expect(btn).toMatchSnapshot();
    });
    it("Expects to find button HTML element in DOM", () => {
        const wrapper = shallow(<Button />);
        expect(wrapper).toHaveLength(1);
    });

    it("Expects to run onClick function when button is pressed in the DOM", () => {
        const mockCallBackClick = jest.fn();
        const wrapper = shallow(
            <Button onClick={mockCallBackClick} className='modalBtn' />
        );
        wrapper.find("button").simulate("click");
        expect(mockCallBackClick.mock.calls.length).toEqual(1);
    });
});
