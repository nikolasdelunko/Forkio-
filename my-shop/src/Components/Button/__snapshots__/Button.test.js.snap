// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component Should render correctly in DOM 1`] = `ShallowWrapper {}`;
